### ISSUE TITLE
[BITV] Global search a11y issues

### ISSUE THREAD

**Issue by PVince81** [nextcloud/server#33742]
These are about the search on the top right.

  * [x] Filter menu
    * [x] (Present on `master`) Needs to use `role="menu" and button role="menuitem"`
    * [x] See <https://inclusive-components.design/menus-menu-buttons/>
  * [x] If search menu is left, it needs to be closed https://github.com/nextcloud/server/pull/35090
  * [x] Single-character search needs to be possible (for single character names in certain languages) https://github.com/nextcloud/server/pull/35091
  * [x] Headings need to have proper contrast, currently 4.2, need to be 4.5 https://github.com/nextcloud/server/pull/35222
    * [x] https://github.com/nextcloud/server/pull/35639

**Comment by Pytal** [comment]
Fixed in https://github.com/nextcloud/server/pull/33826

  * [x] `aria-haspopup="menu"` needs to be removed cause it’s not a menu or any of the other properties
  * [x] Needs a clear visual label instead of placeholder for input
  * [x] Add visually hidden text via aria-describedby below field: "Search starts once you start typing"
  * [x] Highlight for active result needs clear visual border
  * [x] Filter menu
    * [x] Keyboard focus for "Actions" button needs to have a border → needs to use Vue component
  * [x] "Files", "Comments" etc headings need to be h2 instead of pseudo elements ::before
  * [x] Filenames should not be h3, but also just span
  * [x] Sublines should not be h4 either, also just span
  * [x] div around the file preview needs `role="img"` removed, otherwise not accessible

**Comment by Pytal** [comment]
> Headings need to have proper contrast, currently 4.2, need to be 4.5

This is due to the default colors set for `--color-main-background` and `--color-primary-element` in both light and dark modes, is changing `--color-primary` which is the same as `--color-primary-element` an option @jancborchardt? There were probably some discussions on this already buried in the list of issues as well?

https://github.com/nextcloud/server/blob/107aeb36e5435c9eb3c6b597d5a7da4f4909f5a9/apps/theming/css/default.css#L13

**Comment by PVince81** [comment]
were the contrast issues solved indirectly with the redesign of NC 25 ?

**Comment by Pytal** [comment]
> were the contrast issues solved indirectly with the redesign of NC 25 ?

Nada, would suggest using `--color-text-maxcontrast` instead @jancborchardt?

### Dark mode

Before (`--color-primary-element` `#00639A`) | After (`--color-text-maxcontrast` `#8C8C8C`)
--- | ---
![image](https://user-images.githubusercontent.com/24800714/197266293-0e45de63-7092-4e60-b60b-258331068cd6.png) | ![image](https://user-images.githubusercontent.com/24800714/197266323-0dac5127-73d8-4bec-9ef3-debde6d9205f.png)
![image](https://user-images.githubusercontent.com/24800714/197266444-8da1ff63-26f4-4e32-895c-4d7716093d26.png) | ![image](https://user-images.githubusercontent.com/24800714/197266460-1083941e-207c-4c3e-9faa-bf131cf75680.png)

### Light mode
Before (`--color-primary-element` `#00639A`) | After (`--color-text-maxcontrast` `#767676`)
--- | ---
![image](https://user-images.githubusercontent.com/24800714/197266827-eec117da-7da1-43bf-816d-633f871db752.png) | ![image](https://user-images.githubusercontent.com/24800714/197266835-9401c1bc-5a5e-4a71-bf3f-075351d857d4.png)
![image](https://user-images.githubusercontent.com/24800714/197266848-9ad387df-fbcc-4b84-958d-5da3fe95ae9c.png) | ![image](https://user-images.githubusercontent.com/24800714/197266855-821f8f62-f011-4196-ba6c-c8eb96bb95b8.png)

**Comment by PVince81** [comment]
note: will need backport to 25

**Comment by Pytal** [comment]
https://github.com/nextcloud/server/issues/33742#issuecomment-1287318564 done in https://github.com/nextcloud/server/pull/35222

**Comment by Pytal** [comment]
Transferred "aria-live to announce how many elements were found via the search could be nice"
- https://github.com/nextcloud/server/issues/35473
